### PR TITLE
Global rework. Step 3: Gamification mechanics. Data output

### PR DESCRIPTION
**Pull request by mrgambal**
ADD: Every model has its own converter to dict to be further translated into any transmission format like JSON;
ADD: New method `EventModel.get_unread_events(cls, user: User) -> list[Event]`;
ADD: New method `UserStateModel.get_top_users(number: int, sort_by: StateSortingKeys) -> Iterator[UserState]:` to provide a top list of users basing on passed criteria;
ADD: Change signature to `RuleModel.get_actual_rules(skip_ids: list, rule_filter: RuleFilter, is_weekend: bool) -> Iterator[Rule]`;
ADD: New method `FundModel.get_all(filter_by: FundFilterKeys = FundFilterKeys.NO_FILTER) -> Iterator[Fund]` to provide a list of foundations;

CHG: Extend method `RuleModel.get_actual_rules()` with new filtering possibilities: added an option to strictly filter project-related rules;

FIX: Squashed a tiny bug in CLI batch loading procedure definition;
FIX: Slight changes in naming and comments;
FIX: `UserState` equality comparison;
FIX: Revert `mongoengine.connection.register_connection()` call in regards to need to run certain tests w/o launching the whole bunch;

### CODE REVIEW

File: vulyk/control.py [hunk 1]
```diff
@@ -73,9 +73,9 @@ def db():
                                 readable=True,
                                 resolve_path=True),
                 nargs=-1)
-@click.options('--meta', multiple=True,
-               type=(str, str),
-               help='Override meta information for the batch')
+@click.option('--meta', multiple=True,
```
Comment (dchaplinsky) [inline vulyk/control.py]: Love you bro

File: vulyk/blueprints/gamification/models/foundations.py [hunk 2]
```diff
@@ -96,6 +106,28 @@ def find_by_id(cls, fund_id: str) -> Fund:
 
         return result
 
+    @classmethod
+    def get_all(
```
Comment (dchaplinsky) [inline vulyk/blueprints/gamification/models/foundations.py]: Hmm, name doesn't seems right, as `get_all` can apply filters.

File: vulyk/blueprints/gamification/models/events.py [hunk 3]
```diff
@@ -98,12 +99,40 @@ def from_event(cls, event: Event):
             achievements=RuleModel.objects(
                 id__in=[r.id for r in event.achievements]),
             acceptor_fund=None
-                if event.acceptor_fund is None
```
Comment (dchaplinsky) [inline vulyk/blueprints/gamification/models/events.py]: I liked old indent more

Comment (mrgambal) [inline vulyk/blueprints/gamification/models/events.py]: Geez :( That was my autoformatter.

**Pull request merged**